tests/integration/chroot.py

```diff
@@ -0,0 +1,298 @@
+              cmd=i))
+            p = subprocess.run([i], shell=True, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+            rc = p.returncode
+            if rc != 0:
+                logger.error("Command failed.")
+            else:
+                logger.info("Command sucessfully executed.")
+        # Generate chroot sshd directory for run/pid
+        chroot_sshd_run_dir = tmp_dir + "/run/sshd"
+        try:
+            os.mkdir(tmp_dir + "/run/sshd")
+            logger.info("Created directory: {dir}".format(
+              dir=chroot_sshd_run_dir))
+        except OSError:
+            logger.error("Could not create directory: {dir}".format(
+              dir=chroot_sshd_run_dir))
+        # Copy dedicated sshd config to chroot
+        sshd_config_src_file = "integration/misc/sshd_config_integration_tests"
+        chroot_sshd_cfg_dir = tmp_dir + "/etc/ssh/"
+        try:
+            shutil.copyfile(sshd_config_src_file, chroot_sshd_cfg_dir+"sshd_config_integration_tests")
+            logger.info("Copied sshd_config_integration_tests to: {dir}".format(
+              dir=chroot_sshd_cfg_dir))
+        except OSError:
+            logger.error("Could not copy sshd_config_integration_tests to {dir}".format(
+              dir=chroot_sshd_cfg_dir))
+        # Copy ssh / authorized_keys file for root user
+        chroot_root_dir = tmp_dir + "/root/"
+        chroot_ssh_authorized_keys = chroot_root_dir + ".ssh/authorized_keys"
+        ssh_authorized_keys = "/tmp/authorized_keys"
+        self._create_dir(chroot_root_dir+".ssh", 0o600)
+        try:
+            shutil.copyfile("/tmp/authorized_keys", chroot_ssh_authorized_keys)
+            logger.info("Copied authorized_keys to: {dir}".format(
+              dir=chroot_ssh_authorized_keys))
+        except OSError:
+            logger.error("Could not copy authorized_keys to {dir}".format(
+              dir=chroot_ssh_authorized_keys))
+
+
+    def _start_sshd_chroot(self, tmp_dir):
+        """ Start sshd inside the chroot """
+        # Define vars to have it more readable
+        gl_chroot_bin = "/gardenlinux/bin/garden-chroot"
+        chroot = tmp_dir
+        chroot_cmd = "/usr/sbin/sshd -D -f /etc/ssh/sshd_config_integration_tests"
+        # Execute in Popen as background task
+        # while we may perform our integration tests
+        proc_exec = "{chroot_bin} {chroot_env} {chroot_cmd}".format(
+          chroot_bin=gl_chroot_bin,
+          chroot_env=chroot,
+          chroot_cmd=chroot_cmd)
+        p = subprocess.Popen([proc_exec], shell=True)
+        logger.info("Started SSHD in chroot environment.")
+
+
+    def _wait_sshd(self):
+        """ Wait for defined SSH port to become ready in chroot environment """
+        port = self.config["port"]
+        logger.info("Waiting for SSHD in chroot to be ready on tcp/{port}...".format(
+          port=port))
+        cmd_kvm = "ssh-keyscan -p {port} localhost".format(
```

Comment:
I would rename this variable from `cmd_kvm` to `cmd`

Reply:
As discussed in comment https://github.com/gardenlinux/gardenlinux/pull/712#discussion_r833259181, since we will move this method in its own module, I would say we need to adjust this cmd anyways, so if `cmd` is not feasible enough, I would rather call this command `keyscan_cmd` or `ssh_cmd`/`cmd_ssh`